aten/src/ATen/native/quantized/cpu/qnnpack/include/conv_utils.h

```diff
@@ -190,5 +211,160 @@ struct conv_param_t {
       ukernel_type = pytorch_qnnp_ukernel_type_conv;
     }
   }
+
+  /**
+   * @brief Computes the output dimensions given a 2D input.
+   */
+  std::array<size_t, 2> compute_output_dims(
+      std::array<size_t, 2> input_dims) const {
+    std::array<size_t, 2> output_dims;
+    output_dims[0] = compute_output_dimension(input_dims[0],  // width
+                                              pad[1] + pad[3],
+                                              /*adjustment=*/0,
+                                              kernel_dims[0],
+                                              dilation[0],
+                                              subsampling_dims[0],
+                                              /*transpose=*/false);
+    output_dims[1] = compute_output_dimension(input_dims[1],  // height
+                                              pad[0] + pad[2],
+                                              /*adjustment=*/0,
+                                              kernel_dims[1],
+                                              dilation[1],
+                                              subsampling_dims[1],
+                                              /*transpose=*/false);
+    return output_dims;
+  }
+};
+
+struct deconv_param_t {
+  const std::array<uint32_t, 2> kernel_dims; // kernel width, kernel height
+  const std::array<uint32_t, 2> stride_dims; // stride width, height
+  const std::array<uint32_t, 2> dilation_dims; // dilation width, height
+  const std::array<uint32_t, 4> padding_dims; // input padding top, left, bottom, right
+  const std::array<uint32_t, 2> adjustment_dims; // output adjustment
+
+  const uint32_t groups;
+  const size_t input_channels;
+  const size_t output_channels;
+  const uint8_t kernel_zero_point;
+  const float kernel_scale;
+  const uint8_t output_min;
+  const uint8_t output_max;
+
+  // The following are derived parameters
+  enum pytorch_qnnp_ukernel_type ukernel_type; // kernel type based on input params
+  size_t group_input_channels;
+  size_t group_output_channels;
+
+  /**
+   * @brief Constructor for initializing the deconvolution parameters.
+   */
+  deconv_param_t(
```

Comment:
This looks exactly the same as conv_param_t to me. We should just reuse that.

Reply:
They are not exactly the same -- there are additional parameters in the deconv + the output shape is computed differently + the supported kernel type is only conv (no other kernels). Squashing the parameters together will require me to change the signature, which will require to change the files that already use the params. I added a TODO to merge the two, but I doin't think they should be part of this PR.

Reply:
It seems to me that except for this parameter, `adjustment_dims`, everything else is the same. It should be ok to have this present for both conv/deconv. It could be renamed to `deconv_ adjustment_dims`. Can you elaborate a little bit on the change of signature comment and the subsequent change of the files comment you made?
I would like to minimize code duplication, if possible, especially as I implement per channel quant. 
It is ok if you want to do this as separate PR, but I would request you follow-up quickly on the merging TODO, otherwise we will just continue to add more duplicated code especially as I implement per channel quant.

Reply:
Will send a separate PR for the merging